Getstring error Of  index out of range

when  read a xls  with rgb in row.GetCol(i)
i=0 is ok but when i>0 is error 
when I change the xls resave., is all ok



panic: runtime error: index out of range

goroutine 1 [running]:
github.com/shakinm/xlsReader/xls/record.(*LabelSSt).GetString(0xc0000045a0, 0xc000108100, 0x5)
	d:/go/SOURCE/pkg/mod/github.com/shakinm/xls!reader@v0.9.1/xls/record/labelSst.go:46 +0x75
main.xlsread(0x4f9e86, 0x19, 0x4b6f3f, 0xc000081f88, 0xc00001a0b8)
	d:/go/source/src/git.intra/vivi_zhu/mail/dataelk/xlst1/main.go:57 +0xadc

--- a/xls/record/boundSheet.go
+++ b/xls/record/boundSheet.go
@@ -57,7 +57,7 @@ func (r *BoundSheet) Read(stream []byte, vers []byte) {
 
 		fixedStream:=[]byte{r.Cch[0],0x00}
 		fixedStream = append(fixedStream, stream[7:]...)
-		r.stFormat.Read(fixedStream)
+		_ = r.stFormat.Read(fixedStream)
 
 	} else {
 		r.Rgch = make([]byte, int(r.Cch[0]))

--- a/xls/record/format.go
+++ b/xls/record/format.go
@@ -50,7 +50,7 @@ func (r *Format) Read(stream []byte, vers []byte) {
 
 	if bytes.Compare(vers, FlagBIFF8) == 0 {
 		copy(r.ifmt[:], stream[0:2])
-		r.stFormat.Read(stream[2:])
+		_ = r.stFormat.Read(stream[2:])
 	} else {
 		copy(r.ifmt[:], stream[:2])
 		copy(r.cch[:], stream[2:4])

--- a/xls/record/sst.go
+++ b/xls/record/sst.go
@@ -68,7 +68,6 @@ func (s *SST) Read(readType string, grbit byte, prevLen int32) () {
 	for {
 
 		var _rgb structure.XLUnicodeRichExtendedString
-		var rgbSize int
 
 		cch := int(helpers.BytesToUint16(s.RgbSrc[0:2]))
 
@@ -82,28 +81,21 @@ func (s *SST) Read(readType string, grbit byte, prevLen int32) () {
 
 		readType = ""
 
-		if cch >= (len(s.RgbSrc)-3)/(1+int(grbit)) || s.ByteLen > 0 {
+		addBytesLen := (len(s.RgbSrc) - 3) - s.ByteLen
 
-			addBytesLen := (len(s.RgbSrc) - 3) - s.ByteLen
-
-			if cch-s.chLen > addBytesLen/(1+int(grbit)) {
-				s.chLen = s.chLen + addBytesLen/(1+int(grbit))
-				s.ByteLen = s.ByteLen + addBytesLen
-				return
-			} else {
-
-				s.ByteLen = s.ByteLen + (cch-s.chLen)*(1+int(grbit))
-				s.chLen = cch
-				rgbSize = s.ByteLen + 3
-			}
-
+		if cch-s.chLen > addBytesLen/(1+int(grbit)) {
+			s.chLen = s.chLen + addBytesLen/(1+int(grbit))
+			s.ByteLen = s.ByteLen + addBytesLen
+			return
 		} else {
-			rgbSize = cch*(1+int(grbit)) + 3
+			s.ByteLen = s.ByteLen + (cch-s.chLen)*(1+int(grbit))
+			s.chLen = cch
 		}
 
-		_rgb.Read(s.RgbSrc[iOft(&oft, 0):iOft(&oft, uint32(rgbSize))])
+		realSize := _rgb.Read(s.RgbSrc[iOft(&oft, 0):])
+		iOft(&oft, realSize)
 
-		if len(s.RgbSrc) >= int(oft) {
+		if len(s.RgbSrc) >= int(realSize) {
 			s.Rgb = append(s.Rgb, _rgb)
 			s.RgbSrc = s.RgbSrc[int(oft):]
 			s.chLen = 0

--- a/xls/structure/XLUnicodeRichExtendedString.go
+++ b/xls/structure/XLUnicodeRichExtendedString.go
@@ -36,7 +36,7 @@ type XLUnicodeRichExtendedString struct {
 	ExtRst ExtRst
 }
 
-func (s *XLUnicodeRichExtendedString) Read(stream []byte) {
+func (s *XLUnicodeRichExtendedString) Read(stream []byte)   uint32 {
 	var rgbSize uint16
 	oft := uint32(0)
 
@@ -107,6 +107,8 @@ func (s *XLUnicodeRichExtendedString) Read(stream []byte) {
 			}
 		}
 	}
+
+	return oft
 }
 
 func iOft(offset *uint32, inc uint32) uint32 {

